packages/editor/src/systems/RenderMonitorSystem.tsx

```diff
@@ -0,0 +1,130 @@
+The License is based on the Mozilla Public License Version 1.1, but Sections 14
+and 15 have been added to cover use of software over a computer network and 
+provide for limited attribution for the Original Developer. In addition, 
+Exhibit A has been modified to be consistent with Exhibit B.
+
+Software distributed under the License is distributed on an "AS IS" basis,
+WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the
+specific language governing rights and limitations under the License.
+
+The Original Code is Infinite Reality Engine.
+
+The Original Developer is the Initial Developer. The Initial Developer of the
+Original Code is the Infinite Reality Engine team.
+
+All portions of the code written by the Infinite Reality Engine team are Copyright © 2021-2023 
+Infinite Reality Engine. All Rights Reserved.
+*/
+
+import { useEffect } from 'react'
+
+import { defineSystem } from '@ir-engine/ecs/src/SystemFunctions'
+import { PresentationSystemGroup } from '@ir-engine/ecs/src/SystemGroups'
+import { SceneComplexity, SceneComplexityWeights } from '@ir-engine/engine/src/scene/constants/SceneConstants'
+import { getMutableState, useHookstate } from '@ir-engine/hyperflux'
+
+import { useQuery } from '@ir-engine/ecs'
+import { SourceComponent } from '@ir-engine/engine/src/scene/components/SourceComponent'
+import { RenderInfoState } from '@ir-engine/spatial/src/renderer/RenderInfoSystem'
+import { VisibleComponent } from '@ir-engine/spatial/src/renderer/components/VisibleComponent'
+import { LightTagComponent } from '@ir-engine/spatial/src/renderer/components/lights/LightTagComponent'
+import { ResourceState, ResourceType } from '@ir-engine/spatial/src/resources/ResourceState'
+import { useTranslation } from 'react-i18next'
+import { EditorWarningState } from '../services/EditorWarningServices'
+
+type SceneComplexityParams = {
+  vertices: number
+  triangles: number
+  texturesMB: number
+  lights: number
+  drawCalls: number
+  shaderComplexity: number
+}
+
+function calculateSceneComplexity(params: SceneComplexityParams): number {
+  const complexity =
+    SceneComplexityWeights.verticesWeight * params.vertices +
+    SceneComplexityWeights.trianglesWeight * params.triangles +
+    SceneComplexityWeights.texturesMBWeight * params.texturesMB +
+    SceneComplexityWeights.lightsWeight * params.lights +
+    SceneComplexityWeights.drawCallsWeight * params.drawCalls +
+    SceneComplexityWeights.shaderComplexityWeight * params.shaderComplexity
+  return complexity
+}
+
+// count number of lines in vertex and fragment shaders
+function getShaderComplexity(resources: Record<string, any>): number {
+  const countLines = (code?: string): number => (code || '').split('\n').length
+  const totalComplexity = Object.values(resources)
+    .filter((resource) => resource.type === ResourceType.Material)
+    .flatMap((resource) => (Array.isArray(resource.asset) ? resource.asset : [resource.asset]))
```

Comment:
It really isn't necessary to have the ternary operator, nor wrapping resource.asset into a new array, by directly returning resource.asset should work.

Consider the following link as reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap

Reply:
oh, cool ill change it then

Reply:
thats great, thanks for pointing it out

Reply:
applied